Tasks/VsTestV2/inputparser.ts

```diff
@@ -536,14 +536,25 @@ function getTestPlatformPath(inputDataContract : idc.InputDataContract) {
 }
 
 function getVSTestConsolePath(versionLowerLimit : string, versionUpperLimit : string): string {
-    const vswhereTool = tl.tool(path.join(__dirname, 'vswhere.exe'));
+    let vswhereTool = tl.tool(path.join(__dirname, 'vswhere.exe'));
 
     vswhereTool.line(`-version [${versionLowerLimit},${versionUpperLimit}) -latest -products * -requires Microsoft.VisualStudio.PackageGroup.TestTools.Core -property installationPath`);
     let vsPath = vswhereTool.execSync({ silent: true } as tr.IExecSyncOptions).stdout;
     vsPath = utils.Helper.trimString(vsPath);
-    tl.debug('Visual Studio 15.0 or higher installed path: ' + vsPath);
 
     if (!utils.Helper.isNullOrWhitespace(vsPath)) {
+        tl.debug('Visual Studio 15.0 or higher installed path: ' + vsPath);
+        return vsPath;
+    }
+
+    // Look for build tool installation if full VS not present
+    tl.debug('Attempting to find vstest.console from a visual studio build tools installation.');
```

Comment:
why do we need to do this?

You can pass multiple products and pass `-requiresAny`

Reply:
Will cause issues when both build tools as well as VS are present. Safer to split the check